enumerables.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    selfitem.my_each { |x| return false unless yield(x) }
+    true
+  end
```

Comment:
enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
     Failure/Error: expect(true_array.my_all?).to be true_array.all?
     
     LocalJumpError:
       no block given (yield)

enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
     Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
 enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)

 enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)